Sources/PowerSyncGRDB/GRDBPool.swift

```diff
@@ -0,0 +1,171 @@
+import Foundation
+import GRDB
+import PowerSync
+import SQLite3
+
+// The system SQLite does not expose this,
+// linking PowerSync provides them
+// Declare the missing function manually
+@_silgen_name("sqlite3_enable_load_extension")
+func sqlite3_enable_load_extension(
+    _ db: OpaquePointer?,
+    _ onoff: Int32
+) -> Int32
+
+// Similarly for sqlite3_load_extension if needed:
+@_silgen_name("sqlite3_load_extension")
+func sqlite3_load_extension(
+    _ db: OpaquePointer?,
+    _ fileName: UnsafePointer<Int8>?,
+    _ procName: UnsafePointer<Int8>?,
+    _ errMsg: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>?
+) -> Int32
+
+enum PowerSyncGRDBError: Error {
+    case coreBundleNotFound
+    case extensionLoadFailed(String)
+    case unknownExtensionLoadError
+    case connectionUnavailable
+}
+
+struct PowerSyncSchemaSource: DatabaseSchemaSource {
+    let schema: Schema
+
+    func columnsForPrimaryKey(_: Database, inView view: DatabaseObjectID) throws -> [String]? {
+        if schema.tables.first(where: { table in
+            table.viewName == view.name
+        }) != nil {
+            return ["id"]
+        }
+        return nil
+    }
+}
+
+func configurePowerSync(
+    config: inout Configuration,
+    schema: Schema
+) {
+    // Register the PowerSync core extension
+    config.prepareDatabase { database in
+        guard let bundle = Bundle(identifier: "co.powersync.sqlitecore") else {
```

Comment:
I think on watchOS we would have to call the `powersync_init_static()` C function instead, since everything is linked statically there.